deployment/src/strongmind_deployment/autotag.py

```diff
@@ -3,50 +3,98 @@
 from typing import Dict
 from strongmind_deployment.taggable import is_taggable
 import pulumi
-
-# Usage:
-# config = pulumi.Config()
-# register_auto_tags({
-#     'user:Project': pulumi.get_project(),
-#     'user:Stack': pulumi.get_stack(),
-#     'user:Cost Center': config.require('costCenter'),
-# })
-#  Or to take existing defaults:
-# add_standard_billing_tags();
-# 
-# Or to take existing defaults with a couple of additional tags:
-# add_standard_billing_tags({
-#     'Customer': 'Acme Co',
-#     'Service': 'Foobar',
-# })
+import subprocess
 
 ########################
 ###### Entrypoint ######
 
-def get_standard_tags(extra_tags=dict):
+class StandardTags:
+    def __init__(
+        self,
+        extra_tags: dict,
+        project: str = None,
+        environment: str = None,
+        service: str = None,
+        product: str = None,
+        repository: str = None,
+    ):
+        config = pulumi.Config("tags")
+        self.extra_tags = extra_tags
+        self.project = project or pulumi.get_project()
+        self.environment = environment or pulumi.get_stack()
+        self.service = service or config.require("service")
+        self.product = product or config.require("product")
+        # get the git repository from the current git repo if not provided
+        self.repository = repository or config.get("repository") or get_repo_name()
```

Comment:
this just alleviates you from putting one more configuration tag in your pulumi file,  however works if you still want to override it. 
Uses `basename` as you can see below,  so for the `strongmind/shared_infrastrutcure` repo it should just be `shared_infrastructure`.